cinema/tests.py

```diff
@@ -0,0 +1,105 @@
+
+    def test_unauthenticated(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+class AuthenticatedTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(email='<EMAIL>', password='<PASSWORD>')
+        self.client.force_authenticate(user=self.user)
+
+    def movie_list(self):
+        movie = create_a_movie()
+        serializer = MovieSerializer(movie)
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+        
+    def movie_create(self):
+        movie = create_a_movie()
+        res = self.client.post(MOVIE_URL, **movie)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def movie_retrieve(self):
+        movie = create_a_movie()
+        serializer = MovieSerializer(movie)
+        res = self.client.get(MOVIE_URL, {'id': movie.id})
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def movie_update(self):
+        movie = create_a_movie()
+        res = self.client.patch(MOVIE_URL, {'id': movie.id, 'name': 'Test Movie 2',})
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+        
+    def movie_delete(self):
+        movie = create_a_movie()
+        res = self.client.delete(MOVIE_URL, {'id': movie.id})
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AdminTests(AuthenticatedTests):
+    def setUp(self):
+        client = APIClient()
+        user = super(AuthenticatedTests, self).setUp()
+        client.force_authenticate(user=user)
+
+    def movie_list(self):
+        movie = create_a_movie()
+        serializer = MovieSerializer(movie)
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def movie_create(self):
+        movie = create_a_movie()
+        res = self.client.post(MOVIE_URL, **movie)
+        serializer = MovieSerializer(movie)
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(res.data, serializer.data)
+
+    def movie_retrieve(self):
+        movie = create_a_movie()
+        serializer = MovieSerializer(movie)
+        res = self.client.get(MOVIE_URL, {'id': movie.id})
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def movie_update(self):
+        movie = create_a_movie()
+        res = self.client.patch(MOVIE_URL, {'id': movie.id, 'name': 'Test Movie 2', })
+        serializer = MovieSerializer(movie)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
```

Comment:
The `movie_update` method should update the movie object before comparing it with the serializer data. Consider fetching the updated movie from the database after the patch request.